src/charm.py

```diff
@@ -8,18 +8,157 @@
+            "summary": "oathkeeper-operator layer",
+            "description": "pebble config layer for oathkeeper-operator",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "Oathkeeper Operator layer",
+                    "command": f"oathkeeper serve -c {self._oathkeeper_config_path}",
+                    "startup": "enabled",
+                }
+            },
+            "checks": {
+                "alive": {
+                    "override": "replace",
+                    "http": {"url": f"http://localhost:{OATHKEEPER_API_PORT}/health/alive"},
+                },
+                "ready": {
+                    "override": "replace",
+                    "http": {"url": f"http://localhost:{OATHKEEPER_API_PORT}/health/ready"},
+                },
+            },
+        }
+        return Layer(layer_config)
+
+    @property
+    def _oathkeeper_service_is_created(self) -> bool:
+        try:
+            self._container.get_service(self._container_name)
+        except (ModelError, RuntimeError):
+            return False
+        return True
+
+    @property
+    def _kratos_login_url(self) -> str:
+        # This URL is a placeholder, it will be replaced by a juju integration
+        return f"http://kratos.{self.model.name}.svc.cluster.local:4433/self-service/login/browser"
+
+    def _render_access_rules_file(self) -> str:
+        """Render the access rules file."""
+        with open("templates/access-rules.yaml.j2", "r") as file:
+            template = Template(file.read())
+
+        rendered = template.render(
+            kratos_login_url=self._kratos_login_url,
+            return_to="http://default-url.com",
+        )
+        return rendered
+
+    def _render_conf_file(self) -> str:
+        """Render the Oathkeeper configuration file."""
+        with open("templates/oathkeeper.yaml.j2", "r") as file:
+            template = Template(file.read())
+
+        rendered = template.render(
+            kratos_session_url=f"http://kratos.{self.model.name}.svc.cluster.local:4433/sessions/whoami",
+            kratos_login_url=self._kratos_login_url,
+        )
+        return rendered
+
+    def _on_oathkeeper_pebble_ready(self, event: PebbleReadyEvent) -> None:
+        """Event Handler for pebble ready event."""
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Oathkeeper container. Deferring the event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Oathkeeper container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring resources")
+
+        self._container.add_layer(self._container_name, self._oathkeeper_layer, combine=True)
+
+        if not self._oathkeeper_service_is_created:
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting for Oathkeeper service")
```

Comment:
Nitpick from a charming freshman:

- Like above, `Waiting for Oathkeeper service` to be `created` or `ready`?